File: ruby/hamming/hamming.rb
module Hamming
  module_function

  def compute(strand1, strand2)
    if strand1.size != strand2.size
      raise ArgumentError, 'The two strands must have a same length.'
    end
    
    bases1 = strand1.chars
    bases2 = strand2.chars
    bases1.zip(bases2).select { |b1, b2| b1 != b2 }.size
  end
end

module BookKeeping
  VERSION = 3
end

File: scala/hamming/src/main/scala/hamming.scala

object Hamming {
  def compute(strand1: String, strand2: String): Integer = {
    require(strand1.length == strand2.length, "Two strands must have a same length")
    strand1 zip strand2 count { case (b1, b2) => b1 != b2 }
  }
}
